File: Backend/ProjectDashboard/Models/Domain/Team.cs
using System.Collections.Generic;

namespace ProjectDashboard.Models.Domain {
	public class Team {

		public int Id { get; set; }

		public string Name { get; set; }

		public int LeadId { get; set; }
		public User Lead { get; set; }

		public ICollection<TeamMember> Members { get; set; } = new List<TeamMember>();

		public Team() { }

		public Team(string name, User lead) {
			this.Name = name;
			this.Lead = lead;
		}
	}
}

File: Android/app/src/main/java/com/quickdev/projectdashboard/models/domain/Team.kt
package com.quickdev.projectdashboard.models.domain

import androidx.room.*
import com.quickdev.projectdashboard.util.converters.IntListConverter

@Entity(
	tableName = "teams",
	indices = [
		Index(value = ["id"], unique = true)
	]
)
@TypeConverters(value = [IntListConverter::class])
data class Team(
	@PrimaryKey(autoGenerate = false)
	val id: Int = 0,
	val name: String,
	val leadId: Int,
	val memberIds: List<Int>
) {
	/** Variable to be filled in by the repository, not saved in the Room */
	@Ignore
	var lead: User? = null

	/** Variable to be filled in by the repository, not saved in the Room */
	@Ignore
	var members: List<User> = listOf()
}